sopel/config/types.py

```diff
@@ -276,33 +290,64 @@ def parse(self, value):
             multi-line string.
         """
         if "\n" in value:
-            items = [
+            items = (
                 # remove trailing comma
                 # because `value,\nother` is valid in Sopel 7.x
                 item.strip(self.DELIMITER).strip()
-                for item in value.splitlines()]
+                for item in value.splitlines())
```

Comment:
Why change this away from list, only to convert it to list anyway below, in the `return`?

The fallback logic uses `.split()`, which already returns a list. Just seems weirdly redundant.

Reply:
A `(... for ... in ...)` syntax is a generator-expression, it returns a generator (no execution yet, so iteration count is 0), while a `[... for ... in ...]` is a list-expression which creates a list (there was an execution, so the iteration count is N, the length of the list).

I haven't look a the code around, but I assume I reuse `items` for another transformation. If so, we can keep that. If not, then we can safely use a list-expression.

Reply:
So now that I've read the code, this version does:

```
items = (item.strip(...) for item in value.splitlines())  # 0 iteration
items = (self.parse_item(item) for item in items if item)  # 0 iteration
if self.strip:
    return [item.strip() for item in items]  # N iterations
# else:
return list(items) # N iterations
```

So in total: N iterations, N being the number of channels.

On the other side, if we change these generator-expression into list-expression, we will have:


```
items = [item.strip(...) for item in value.splitlines()]  # N iterations
items = [self.parse_item(item) for item in items if item]  # +N iterations
if self.strip:
    return [item.strip() for item in items]  # +N iterations
# else:
return items # 0 iteration
```

At best, we have 2N iterations, at worst 3N iterations. It's nothing, in term of optimization, because the data is so small. Yet, I've this habit of using generator-expression and `yield` syntax everywhere I can.

Reply:
Fair enough. Guess I've outed myself through this conversation, revealing that I never use generators. 😁